pkg/resourcemanager/controller/managedresource/reconciler.go

```diff
@@ -270,12 +275,19 @@ func (r *reconciler) reconcile(ctx context.Context, mr *resourcesv1alpha1.Manage
 		}
 	}
 
+	// get the current checksum of referenced secret data stored as annotation.
+	currentChecksum := ""
+	annotations := mr.ObjectMeta.GetAnnotations()
+	if annotations != nil && metav1.HasAnnotation(mr.ObjectMeta, "checksum/secret-data") {
+		currentChecksum = currentChecksum + annotations["checksum/secret-data"]
+	}
```

Comment:
```suggestion
	currentChecksum := mr.ObjectMeta.GetAnnotations()["checksum/data"]
```

should work and be safe?

Reply:
You should also compute a checksum of `secretChecksum` again. Otherwise, if the `ManagedResource` has many `secretRef`s this will become very long. 

```golang
secretChecksum = utils.ComputeChecksum(secretChecksum)
```

Reply:
> 

I thought about this but wasn't sure about the limit for the length of annotation.

Reply:
The limit is quite high, still we don't want to unnecessary flood etcd with unneeded data